### ISSUE TITLE
Soft assignment different from paper 

### ISSUE THREAD

**Issue by bowenxin** [vlukiyanov/pt-dec#13]
Hi, vlukiyanov, 

Thanks for your great pytorch implementation of DEC. However, I noticed there is a place where your implementation different from the original paper. 

In the cluster.py class, ClusterAssignment.forward() function, your implementation is that: 
`def forward(self, batch: torch.Tensor) :

        norm_squared = torch.sum((batch.unsqueeze(1) - self.cluster_centers) ** 2, 2)
        numerator = 1.0 / (1.0 + (norm_squared / self.alpha))
        power = float(self.alpha + 1) / 2
        numerator = numerator**power
        return numerator / torch.sum(numerator, dim=1, keepdim=True)`

However, in the original paper, there is a "minus" in the power term. 
![image](https://user-images.githubusercontent.com/10879680/58846558-ec6f3800-86c2-11e9-83f1-0d430f1ac722.png)

Would you please explain why this is different from the original paper? 

Regards,
Bowen

**Comment by vlukiyanov** [comment]
Hey, 

As far as I can see  `numerator = 1.0 / (1.0 + (norm_squared / self.alpha))` negates the power, so essentially the power is `-float(self.alpha + 1) / 2`. Have a look at the tests, I think they make sense calculating by hand https://github.com/vlukiyanov/pt-dec/blob/master/tests/test_cluster.py, but I could always be wrong so would be good to know if the above now makes sense.

Vlad
